src/ui.jsx

```diff
@@ -0,0 +1,392 @@
+/**
+ * ui.jsx
+ *
+ * A file describing the UI of the application using React. A lot of the state accesses occur on the state property
+ * which apparently isn't the best practice but it gets rid of lots of unnecessary re-renders.
+ *
+ */
+"use strict";
+/*eslint-disable no-console, no-underscore-dangle*/
+
+import React from "react";
+
+import { executeInstruction, getFancyPositionInformation } from "./robot.js";
+import { parseInstructions, prepareRobots } from "./instructionreader.js";
+
+export default class MartianRobots extends React.Component {
+
+    render() {
+
```

Comment:
Try not to add whitespace where it isn't necessary, I don't think it makes it any more readable.